cli/src/pcluster/config/cluster_config.py

```diff
@@ -294,13 +297,37 @@ def _register_validators(self):
             self._register_validator(KmsKeyIdEncryptedValidator, kms_key_id=self.kms_key_id, encrypted=self.encrypted)
 
 
-class SharedFsx(Resource):
+class BaseSharedFsx(Resource):
+    """Represent the shared FSX resource."""
+
+    def __init__(self, mount_dir: str, name: str):
+        super().__init__()
+        self.mount_dir = Resource.init_param(mount_dir)
+        self.name = Resource.init_param(name)
+        self.shared_storage_type = SharedStorageType.FSX
+        self.__file_system_data = None
+
+    def _register_validators(self):
+        self._register_validator(SharedStorageNameValidator, name=self.name)
```

Comment:
Is it possible to test the inclusion of SharedStorageNameValidator (see codecov comment below)?

Reply:
Done